OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -201,11 +215,31 @@ protected override void Created(Actor self)
 		{
 			notifyCustomLayerChanged = self.TraitsImplementing<INotifyCustomLayerChanged>().ToArray();
 			notifyVisualPositionChanged = self.TraitsImplementing<INotifyVisualPositionChanged>().ToArray();
+			notifyMoving = self.TraitsImplementing<INotifyMoving>().ToArray();
 			notifyFinishedMoving = self.TraitsImplementing<INotifyFinishedMoving>().ToArray();
 
 			base.Created(self);
 		}
 
+		void ITick.Tick(Actor self)
+		{
+			// Check if actor has moved horizontally. If yes, enable flag, if no, disable flag.
+			if (oldPos != CenterPosition)
+				CurrentMovementTypes |= MovementType.Horizontal;
+			else
+				CurrentMovementTypes &= ~MovementType.Horizontal;
+
+			oldPos = CenterPosition;
+
+			// Check if actor has changed facing (turned). If yes, enable flag, if no, disable flag.
+			if (oldFacing != Facing)
+				CurrentMovementTypes |= MovementType.Turn;
```

Comment:
As above, setting each flag in turn may cause the notification to be fired multiple times with different values in the same tick.  It would be best to set both flags in a temporary variable first, then assign that to `CurrentMovementTypes` once.